lib/fetch/util.js

```diff
@@ -844,9 +841,26 @@ function makeIterator (iterator, name, kind, keyIndex = 0, valueIndex = 1) {
     configurable: true
   })
 
-  // esIteratorPrototype needs to be the prototype of iteratorObject
-  // which is the prototype of an empty object. Yes, it's confusing.
-  return Object.create(iteratorObject)
+  /**
+   * @param {unknown} target
+   * @param {'key'|'value'|'key+value'} kind
+   */
+  return function (target, kind) {
+    // esIteratorPrototype needs to be the prototype of iteratorObject
+    // which is the prototype of an empty object. Yes, it's confusing.
+    const iterator = Object.create(iteratorObject)
+    Object.defineProperty(iterator, kInternalObject, {
+      value: {
+        target,
+        kind,
+        index: 0
+      },
+      writable: false,
+      enumerable: false,
+      configurable: true
+    })
+    return iterator
+  }
```

Comment:
can we somehow avoid Object.create and Object.defineProperty? 

Maybe using class?

Usually those functions are slow.

Reply:
I agree. Let's optimize in this way. https://github.com/tsctx/webidl-dfn-iterator-prototype-object/commit/3eee31df91dd0cc650fc9507785b303f0bdabc0e

Reply:
Done. https://github.com/nodejs/undici/pull/2743/commits/19a7d8d17ba38e7999112d9cf999dfdb6ded0dc7

Reply:
nice